Makefile

```diff
@@ -0,0 +1,33 @@
+# Makefile setup for docker repository
+# Adapted from: https://raw.githubusercontent.com/clemblanco/pod-point-coding-test/master/Makefile
+
+help: ## This help.
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+.DEFAULT_GOAL := help
+
+# If the first argument is "run"...
+ifeq (run,$(firstword $(MAKECMDGOALS)))
+  # use the rest as arguments for "run"
+  RUN_ARGS := $(wordlist 2,$(words $(MAKECMDGOALS)),$(MAKECMDGOALS))
+  # ...and turn them into do-nothing targets
+  $(eval $(RUN_ARGS):;@:)
+endif
+
+# If the first argument is "test"...
+ifeq (test,$(firstword $(MAKECMDGOALS)))
```

Comment:
This code is duplicated as it allows us to pass options to the command in switch we are running. I couldn't for the life of me make it work as one single bit of code. I'm sure there is a better way of doing this but for the sake of my sanity I'm losing it as is but open to the idea of changing it, if you have a better way of handling it.